.github/workflows/functional-tests.yml

```
@@ -4,21 +4,58 @@ on:
 
     steps:
+      # In a called workflow the github context belongs to the CALLER, so a bare
+      # checkout would clone the calling repository and leave no test suite to
+      # run. job.workflow_repository / job.workflow_sha point at the repo and
+      # commit this workflow file itself came from, which is what we want there.
+      # inputs.environment is only set when we are called, so direct push and
+      # pull_request runs keep checking out github.repository at github.sha
+      # exactly as before.
       - uses: actions/checkout@v4
+        with:
+          repository: ${{ inputs.environment && job.workflow_repository || github.repository }}
```

actionlint (1.7.12)

[error] property "workflow_repository" is not defined in object type {check_run_id: number; container: {id: string; network: string}; services: {string => {id: string; network: string; ports: {string => string}}}; status: string}

(expression)